File: spring-web-java/src/main/java/spring/web/java/domain/member/dto/MemberResponse.java
package spring.web.java.domain.member.dto;

import java.time.OffsetDateTime;

import spring.web.java.domain.member.Member;
import spring.web.java.global.common.DateTimeUtils;

public record MemberResponse(
	Long id,
	String account,
	String name,
	String city,
	String street,
	String zipcode,
	Member.Role role,
	OffsetDateTime createdAt
) {
	public MemberResponse(Member member) {
		this(
			member.getId(),
			member.getAccount(),
			member.getName(),
			member.getAddress().getCity(),
			member.getAddress().getStreet(),
			member.getAddress().getZipcode(),
			member.getRole(),
			DateTimeUtils.toOffsetDateTime(member.getCreatedAt())
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/member/dto/MemberResponse.kt
package spring.web.kotlin.domain.member.dto

import spring.web.kotlin.domain.member.Member
import spring.web.kotlin.global.common.toOffsetDateTime
import java.time.OffsetDateTime

data class MemberResponse(
    val id: Long,
    val account: String,
    val name: String,
    val city: String,
    val street: String,
    val zipcode: String,
    val role: Member.Role,
    val createdAt: OffsetDateTime,
) {
    constructor(member: Member) : this(
        id = member.id!!,
        account = member.account,
        name = member.name,
        city = member.address.city,
        street = member.address.street,
        zipcode = member.address.zipcode,
        role = member.role,
        createdAt = member.createdAt!!.toOffsetDateTime(),
    )
}
